### ISSUE TITLE
Create REST API

### ISSUE THREAD

**Issue by frankscholten** [ContainerSolutions/minimesos#249]
The goal of this feature is to open up minimesos to other languages. Therefore let's create a REST API and refactor the code so that MesosCluster uses the REST API internally.

Once that is in place we can add clients for other languages as separately managed projects.

**Comment by frankscholten** [comment]
* Added rest-api module with Dockerfile which starts a simple SparkJava server. See http://sparkjava.com
* Added modules minimesos-java which is the Java client which will connect to the REST API
* Next steps are expanding the MesosClusterTest in minimesos-java so that it runs the rest-api on start and talks to it for any subsequent commands.
 Once all commands are covered in minimesos-java we can look into connecting the cli and refactoring the rest of the code.

**Comment by alex-glv** [comment]
I think we need a blueprint that we can do a collaborative edit on and solidify the API design

**Comment by frankscholten** [comment]
Let me create a checkvist list

**Comment by frankscholten** [comment]
https://checkvist.com/checklists/544419/accept?secret=4C5qkDpgT1dplv

**Comment by alex-glv** [comment]
@frankscholten what state is the API at? were you working on a branch, do you keep it up to date?

**Comment by frankscholten** [comment]
The above commit is the only thing. Didn't touch it after that

**Comment by frankscholten** [comment]
Closing as this code is very much out of date.

Created a new issue https://github.com/ContainerSolutions/minimesos/issues/450
